File: src/viewer/math/GLSphericalCoordinates.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package viewer.math;

import com.hackoeur.jglm.Vec3;
import org.apache.commons.math3.geometry.euclidean.threed.SphericalCoordinates;
import org.apache.commons.math3.geometry.euclidean.threed.Vector3D;

/**
 *
 * @author palasjiri
 */
public class GLSphericalCoordinates extends SphericalCoordinates{

    public GLSphericalCoordinates(Vector3D v) {
        super(new Vector3D(v.getZ(), v.getX(), v.getY()));
    }
    
    public GLSphericalCoordinates(Vec3 v){        
        super(new Vector3D(v.getX(), v.getY(), v.getZ()));
    }

    public GLSphericalCoordinates(double radius, double theta, double phi) {
        super(radius, theta, phi);
    }
   
    @Override
    public Vector3D getCartesian() {
        Vector3D v = super.getCartesian();
        return new Vector3D(v.getY(), v.getZ(), v.getX());
    }
    
    public Vec3 getCartesianGLM(){
        Vector3D v = super.getCartesian();
        return new Vec3( (float) v.getY(), (float) v.getZ(), (float) v.getX());
    }
    
    
}

File: src/main/kotlin/palasjir/viewer/coordinates/GLSphericalCoordinates.kt
package palasjir.viewer.coordinates

import com.hackoeur.jglm.Vec3
import org.apache.commons.math3.geometry.euclidean.threed.SphericalCoordinates
import org.apache.commons.math3.geometry.euclidean.threed.Vector3D

class GLSphericalCoordinates : SphericalCoordinates {

    constructor(v: Vec3) : super(Vector3D(v.z.toDouble(), v.x.toDouble(), v.y.toDouble()))

    constructor(radius: Double, theta: Double, phi: Double) : super(radius, theta, phi)

    val cartesian: Vec3
        get() {
            val v = super.getCartesian()
            return Vec3(v.y.toFloat(), v.z.toFloat(), v.x.toFloat())
        }
}
